client/src/features/Catalogue/components/CatalogFilterDuration/CatalogFilterDuration.js

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect } from 'react';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+import Grid from '@mui/material/Unstable_Grid2';
+import { FilterAccordion, OutlinedToggleButton } from '..';
+import { getProducts } from '../../../../store/slices/catalogueSlice/catalogueSlice';
+
+
+const CatalogFilterDuration = () => {
+  const dispatch = useDispatch();
+  const products = useSelector((state) => state.catalogue.products, shallowEqual);
+
+  const sorting = () => {
+    const duration = [...new Set(products.map((el) => el.duration))];
+
+    const hours = duration
+      .filter((el) => el.indexOf('hour') !== -1)
+      .sort((a, b) => a.match(/\d+/)[0] - b.match(/\d+/)[0]);
+
+    const days = duration
+      .filter((el) => el.indexOf('day') !== -1)
+      .sort((a, b) => a.match(/\d+/)[0] - b.match(/\d+/)[0]);
+
+    return [...hours, ...days];
+  };
```

Comment:
this is so complicated, better ask the database owner to just change the data format of the duration property to a more comfortable format to work with in your filter.

Reply:
you are right, but I still cannot determine which data format is more convenient and easier to work with